src/iron.rs

```diff
@@ -111,89 +128,139 @@ impl<H: Handler> Iron<H> {
     /// Kick off the server process using the HTTP protocol.
     ///
     /// Call this once to begin listening for requests on the server.
-    /// This consumes the Iron instance, but does the listening on
-    /// another task, so is not blocking.
-    ///
-    /// The thread returns a guard that will automatically join with the parent
-    /// once it is dropped, blocking until this happens.
-    pub fn http<A>(self, addr: A) -> HttpResult<Listening>
+    pub fn http<A>(self, addr: A) -> HttpResult<()>
         where A: ToSocketAddrs
     {
-        HttpListener::new(addr).and_then(|l| self.listen(l, Protocol::http()))
+        let addr = addr.to_socket_addrs()?.next().unwrap();
+
+        let core = Core::new().unwrap();
+        let handle = core.handle();
+
+        let sock = TcpListener::bind(&addr, &handle).unwrap().incoming();
+
+        return self.listen(sock, addr, Protocol::http(), core, handle);
     }
 
     /// Kick off the server process using the HTTPS protocol.
     ///
     /// Call this once to begin listening for requests on the server.
-    /// This consumes the Iron instance, but does the listening on
-    /// another task, so is not blocking.
-    ///
-    /// The thread returns a guard that will automatically join with the parent
-    /// once it is dropped, blocking until this happens.
-    pub fn https<A, S>(self, addr: A, ssl: S) -> HttpResult<Listening>
-        where A: ToSocketAddrs,
-              S: 'static + SslServer + Send + Clone
+    #[cfg(feature = "ssl")]
```

Comment:
This shouldn't lock us into native-tls, just tokio-tls.  This still uses the tokio-tls trait TlsAcceptorExt, which should be implementable by any TLS crate (see: https://tokio-rs.github.io/tokio-tls/tokio_tls/trait.TlsAcceptorExt.html ).

If the tokio-tls dependency is unwanted, I could make our own version of that trait.  I think that implementation can even make using something implementing TlsAcceptorExt work.

Unfortunately, it does drag in a native-tls dependency to require tokio-tls.